base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -1582,9 +1582,16 @@ fn handle_reorg<T: BlockchainBackend>(
         .height -
         1;
     let removed_blocks = reorganize_chain(db, block_validator, fork_height, reorg_chain)?;
-    if removed_blocks.is_empty() {
-        Ok(BlockAddResult::Ok)
-    } else {
+    let num_removed_blocks = removed_blocks.len();
+    let num_added_blocks = added_blocks.len();
+
+    if reorg_required(num_removed_blocks, num_added_blocks) {
```

Comment:
I agree it's a bit verbose but it is explicit. Instead of creating new variables I'll just pass in the lengths. I respectfully disagree about the function though, given that there's currently no integration test for this the `handle_reorg` function (afaik). At least this way the intent is clear, explicit, and has a (admittedly simple) test to catch a regression in case the `reorg_required` function is changed. Deferring to @mikethetike - keep the `reorg_required` function and test for now?

Reply:
Fair enough, it was a nit and as you say perhaps the verbosity is justified here.